src/ILCompiler.Compiler/src/Compiler/PreInitFieldInfo.cs

```diff
@@ -145,6 +150,10 @@ public static List<PreInitFieldInfo> GetPreInitFieldInfos(TypeDesc type)
                 if (!field.IsStatic)
                     continue;
 
+                bool isValueType = IsSupportedValueType(field.FieldType);
```

Comment:
I would kind of prefer if this was keying of off `FieldDesc.HasGCStaticBase`. This is what ultimately decides where the field gets placed. Trying to replicate the logic here is bound to have bugs.

Also the "wantValueType" parameter is kind of odd. We should name it for what it is: the preinitialized data for the GC statics or non-GC statics.